### PR TITLE
Sample model generation script for on-device-training

### PR DESCRIPTION
**Pull request by baijumeswani**
This pull request adds a python script that can generate a training model (forward+loss+gradients) for a forward only onnx model, along with the optimizer model.
The outputs of the script are:
- Training onnx model.
- Eval onnx model
- Optimizer onnx model
- Checkpoint to be used in online training.

### CODE REVIEW

**Review by pengwa — COMMENTED** [review]
is this requested by someone from PM side?

File: orttraining/orttraining/python/training/onnxblock/sample/model_generation.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import onnx
+
+import onnxruntime.training.onnxblock as onnxblock
+
+
+class MyTrainingModel(onnxblock.TrainingModel):
+    def __init__(self):
+        super().__init__()
+        self._loss = {
+            "MSELoss": onnxblock.loss.MSELoss(),
```
Comment (pengwa) [inline orttraining/orttraining/python/training/onnxblock/sample/model_generation.py:17]: if this is a sample code, I think reader would be confused why offline requires 3 loss functions? In most cases, I think single loss is needed, right?
